app/connectors/MessageConnector.scala

```diff
@@ -163,14 +186,20 @@ class MessageConnector @Inject() (
                                 |X-Message-Sender: ${getHeader("X-Message-Sender", details.url)}
                                 |Accept: ${getHeader("Accept", details.url)}
                                 |CustomProcessHost: ${getHeader("CustomProcessHost", details.url)}
+                                |Timestamp (UTC): ${DateTimeFormatter.ISO_DATE_TIME.format(ZonedDateTime.now(ZoneOffset.UTC))}
                                 |""".stripMargin
 
       details.circuitBreaker
         .withCircuitBreaker(
-          {
-            http
-              .POSTString[HttpResponse](details.url, xml.toString)
-              .map { result =>
+          httpv2
+            .post(url"${details.url}")
+            .withBody(payload)
+            .transform(
+              req => req.withRequestTimeout(timeout)
+            )
+            .execute[HttpResponse]
+            .map {
+              result =>
```

Comment:
This is the crux of the change. I'm not convinced we should determine the size of the message like that -- the alternative is passing the content length header value from the original request that triggered this, and I'm not sure about that.

The problem is that because strings are encoded in UTF-16BE, but we would want to send the string in UTF-8, simply getting the length of the string isn't totally accurate, but it might be enough for what we want?